diff -up libopenshot-0.2.7/src/FFmpegWriter.cpp.omv~ libopenshot-0.2.7/src/FFmpegWriter.cpp
--- libopenshot-0.2.7/src/FFmpegWriter.cpp.omv~	2022-01-16 00:50:57.170090521 +0100
+++ libopenshot-0.2.7/src/FFmpegWriter.cpp	2022-01-16 01:00:32.737451159 +0100
@@ -124,7 +124,7 @@ void FFmpegWriter::Open() {
 // auto detect format (from path)
 void FFmpegWriter::auto_detect_format() {
 	// Auto detect the output format from the name. default is mpeg.
-	fmt = av_guess_format(NULL, path.c_str(), NULL);
+	fmt = const_cast<AVOutputFormat*>(av_guess_format(NULL, path.c_str(), NULL)); // FIXME const_cast is evil
 	if (!fmt)
 		throw InvalidFormat("Could not deduce output format from file extension.", path);
 
@@ -1245,7 +1245,7 @@ AVStream *FFmpegWriter::add_video_stream
 	st->avg_frame_rate = av_inv_q(c->time_base);
 	st->time_base.num = info.video_timebase.num;
 	st->time_base.den = info.video_timebase.den;
-#if (LIBAVFORMAT_VERSION_MAJOR >= 58)
+#if (LIBAVFORMAT_VERSION_MAJOR == 58)
 	#pragma GCC diagnostic push
 	#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 	st->codec->time_base.num = info.video_timebase.num;
@@ -2027,7 +2027,7 @@ bool FFmpegWriter::write_video_packet(st
 		pkt.flags |= AV_PKT_FLAG_KEY;
 		pkt.stream_index = video_st->index;
 		pkt.data = (uint8_t *) frame_final->data;
-		pkt.size = sizeof(AVPicture);
+		pkt.size = abs(frame_final->height*frame_final->linesize[0]);
 
 		// Set PTS (in frames and scaled to the codec's timebase)
 		pkt.pts = video_timestamp;
diff -up libopenshot-0.2.7/src/FFmpegWriter.h.omv~ libopenshot-0.2.7/src/FFmpegWriter.h
