diff -uraN libopenshot-0.4.0/CMakeLists.txt omv-libopenshot-0.4.0/CMakeLists.txt
--- libopenshot-0.4.0/CMakeLists.txt	2024-12-21 23:27:30.000000000 +0100
+++ omv-libopenshot-0.4.0/CMakeLists.txt	2024-12-23 18:09:18.589065011 +0100
@@ -97,7 +97,7 @@
 endif()
 
 #### Set C++ standard level
-set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
